plugins/filter/json_patch.py

```diff
@@ -0,0 +1,176 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) Stanislav Meduna (@numo68)
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, annotations, division, print_function
+from json import loads
+from typing import TYPE_CHECKING
+from ansible.errors import AnsibleFilterError
+from ansible.module_utils.common.text.converters import to_native
+
+__metaclass__ = type  # pylint: disable=C0103
+
+if TYPE_CHECKING:
+    from typing import Any, Callable, Union
+
+try:
+    import jsonpatch
+
+except ImportError as exc:
+    HAS_LIB = False
+    JSONPATCH_IMPORT_ERROR = exc
+else:
+    HAS_LIB = True
+    JSONPATCH_IMPORT_ERROR = None
+
+OPERATIONS_AVAILABLE = ["add", "copy", "move", "remove", "replace", "test"]
+OPERATIONS_NEEDING_FROM = ["copy", "move"]
+OPERATIONS_NEEDING_VALUE = ["add", "replace", "test"]
+
+
+class FilterModule:
+    """Filter plugin."""
+
+    def check_json_object(
+        self, filter_name: str, object_name: str, inp: Union[str, list, dict]
```

Comment:
Why not use `Any` here? After all you're verifying the type in the function.
```suggestion
        self, filter_name: str, object_name: str, inp: Any
```